pkg/autostart/autostart.go

```diff
@@ -0,0 +1,119 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); err != nil && !errors.Is(err, os.ErrNotExist) && !errors.Is(err, os.ErrExist) {
```

Comment:
I don't think `os.Stat` will ever return `os.ErrExist`. If the file at the given path exists, it will return `nil`.

`os.Mkdir` is the only function I can think of right now that will return `os.ErrExist` if the directory already exists. Are there any others?

```suggestion
	if _, err := os.Stat(unitPath); err != nil && !errors.Is(err, os.ErrNotExist) {
```